gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.java

```diff
@@ -126,48 +118,7 @@ public final void setProductType(ProductType productType) {
         return productDependencies;
     }
 
-    public final void productDependency(String mavenCoordVersionRange) {
-        productDependency(mavenCoordVersionRange, null);
-    }
-    public final void productDependency(String mavenCoordVersionRange, String recommendedVersion) {
-        Matcher matcher = MAVEN_COORDINATE_PATTERN.matcher(mavenCoordVersionRange);
-        Preconditions.checkArgument(matcher.matches(), "String '%s' is not a valid maven coordinate. "
-                    + "Must be in the format 'group:name:version:classifier@type', where ':classifier' and '@type' are "
-                    + "optional.", mavenCoordVersionRange);
-        String minVersion = matcher.group("version");
-        productDependencies.add(new ProductDependency(
-                matcher.group("group"),
-                matcher.group("name"),
-                minVersion,
-                generateMaxVersion(minVersion),
-                recommendedVersion));
-    }
-
-    public final void productDependency(String dependencyGroup, String dependencyName, String minVersion) {
-        productDependency(dependencyGroup, dependencyName, minVersion, null, null);
-    }
-
-    public final void productDependency(
-            String dependencyGroup, String dependencyName, String minVersion, String maxVersion) {
-        productDependency(dependencyGroup, dependencyName, minVersion, maxVersion,  null);
-    }
-
-    public final void productDependency(
-            String dependencyGroup,
-            String dependencyName,
-            String minVersion,
-            String maxVersion, String recommendedVersion) {
-        productDependencies.add(new ProductDependency(
-                dependencyGroup,
-                dependencyName,
-                minVersion,
-                maxVersion == null
-                        ? generateMaxVersion(minVersion)
-                        : maxVersion,
-                recommendedVersion));
-    }
-
-    public final void productDependency(Closure closure) {
+    public final void productDependency(@DelegatesTo(ProductDependency.class) Closure closure) {
```

Comment:
Not sure we want to remove these methods, just seems like additional churn

Reply:
Sure, we can keep them and specify defaults instead of null for recommendedVersion